File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CommonDialogs/ColorDialog/Functionality/AllowFullOpenPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.CommonDialogs.ColorDialog.Functionality
{
    public partial class AllowFullOpenPage : UserControl
    {
        /// <summary>
        /// Indicates whether button was click for backgroud color change 
        /// </summary>
        private bool _isBackColorLabelChanged = false;

        public AllowFullOpenPage()
        {
            InitializeComponent();
        }

        private void mobjChangeBackColorButton_Click(object sender, EventArgs e)
        {
            //Set represented Label backgroud color a initial color for ColorDialog
            this.mobjDemoColorDialog.Color = this.mobjColorLabel.BackColor;
            //Changed value of flag, that indicates whether button was click for 
            //backgroud color change, to true
            _isBackColorLabelChanged = true;
            //Show ColorDialog
            this.mobjDemoColorDialog.ShowDialog();
        }

        private void mobjChangeForeColorButton_Click(object sender, EventArgs e)
        {
            //Set represented Label foreground color a initial color for ColorDialog
            this.mobjDemoColorDialog.Color = this.mobjColorLabel.ForeColor;
            //Changed value of flag, that indicates whether button was click for 
            //foreground color change, to false
            _isBackColorLabelChanged = false;
            //Show ColorDialog
            this.mobjDemoColorDialog.ShowDialog();
        }

        private void mobjDemoColorDialog_Closed(object sender, EventArgs e)
        {
            //Check dialog result for closed ColorDialog
            if (this.mobjDemoColorDialog.DialogResult == DialogResult.OK)
            {
                //Check whether backgroud color of represented Label should be changed
                if (_isBackColorLabelChanged)
                {
                    //Set backgroud color of represented Label to selected color
                    this.mobjColorLabel.BackColor = this.mobjDemoColorDialog.Color;
                }
                else
                {
                    //Set foregroud color of represented Label to selected color
                    this.mobjColorLabel.ForeColor = this.mobjDemoColorDialog.Color;
                }
            }
           
        }

    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CommonDialogs/ColorDialog/Functionality/AllowFullOpenPage.vb
Namespace CompanionKit.Controls.CommonDialogs.ColorDialog.Functionality

	Public Class AllowFullOpenPage

        Private _isBackColorLabelChanged As Boolean = False


		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        Private Sub mobjChangeBackColorButton_Click(sender As Object, e As EventArgs) Handles mobjChangeBackColorButton.Click
            'Set represented Label backgroud color a initial color for ColorDialog
            Me.mobjDemoColorDialog.Color = Me.mobjColorLabel.BackColor
            'Changed value of flag, that indicates whether button was click for 
            'backgroud color change, to true
            _isBackColorLabelChanged = True
            'Show ColorDialog
            Me.mobjDemoColorDialog.ShowDialog()
        End Sub

        Private Sub mobjChangeForeColorButton_Click(sender As Object, e As EventArgs) Handles mobjChangeForeColorButton.Click
            'Set represented Label foreground color a initial color for ColorDialog
            Me.mobjDemoColorDialog.Color = Me.mobjColorLabel.ForeColor
            'Changed value of flag, that indicates whether button was click for 
            'foreground color change, to false
            _isBackColorLabelChanged = False
            'Show ColorDialog
            Me.mobjDemoColorDialog.ShowDialog()
        End Sub

        Private Sub mobjDemoColorDialog_Closed(sender As Object, e As EventArgs) Handles mobjDemoColorDialog.Closed
            'Check dialog result for closed ColorDialog
            If Me.mobjDemoColorDialog.DialogResult = DialogResult.OK Then
                'Check whether backgroud color of represented Label should be changed
                If _isBackColorLabelChanged Then
                    'Set backgroud color of represented Label to selected color
                    Me.mobjColorLabel.BackColor = Me.mobjDemoColorDialog.Color
                Else
                    'Set foregroud color of represented Label to selected color
                    Me.mobjColorLabel.ForeColor = Me.mobjDemoColorDialog.Color
                End If
            End If

        End Sub
	End Class

End Namespace